samza-core/src/main/java/org/apache/samza/clustermanager/AbstractContainerAllocator.java

```diff
@@ -121,18 +146,101 @@ public void run() {
 
         Thread.sleep(allocatorSleepIntervalMs);
       } catch (InterruptedException e) {
-        log.warn("Got InterruptedException in AllocatorThread.", e);
+        LOG.warn("Got InterruptedException in AllocatorThread.", e);
         Thread.currentThread().interrupt();
       } catch (Exception e) {
-        log.error("Got unknown Exception in AllocatorThread.", e);
+        LOG.error("Got unknown Exception in AllocatorThread.", e);
       }
     }
   }
 
   /**
    * Assigns resources received from the cluster manager to processors.
+   *
+   * During the run() method, the thread sleeps for allocatorSleepIntervalMs ms. It then invokes assignResourceRequests,
+   * and tries to allocate any unsatisfied request that is still in the request queue {@link ResourceRequestState})
+   * with allocated resources.
+   * When {@code hostAffinityEnabled} is disabled, all allocated resources are buffered in the list keyed by "ANY_HOST".
+   * When {@code hostAffinityEnabled} is enabled, all allocated resources are buffered in the list keyed by "preferredHost
```

Comment:
When host-affinity is enabled the allocated resources are keyed by hostName; in case a resourceRequest with this hostName was never made, it is keyed under ANY.

When host-affinity is disabled, all resources are keyed under ANY.